sunpy/map/mapbase.py

```diff
@@ -428,36 +443,50 @@ def observatory(self):
 
     @property
     def xrange(self):
-        """Return the X range of the image from edge to edge."""
-        # TODO: This should be reading from the WCS object
-        xmin = self.center.x - self.dimensions[0] / 2. * self.scale.x
-        xmax = self.center.x + self.dimensions[0] / 2. * self.scale.x
-        return u.Quantity([xmin, xmax])
+        """
+        Return the Longitude (X) range of the image from edge to edge.
+
+        *Note:* This does not imply that the values are linear along this
+         range, this just calculates the longitude values for the first and
+         last pixel.
+        """
+        cmin = self.pixel_to_data(0*u.pix, 0*u.pix)
+        lon_min, lat_min = self._get_lon_lat(cmin)
+        cmax = self.pixel_to_data(*self.dimensions)
+        lon_max, lat_max = self._get_lon_lat(cmax)
+        return u.Quantity([lon_min, lon_max])
 
     @property
     def yrange(self):
-        """Return the Y range of the image from edge to edge."""
-        # TODO: This should be reading from the WCS object
-        ymin = self.center.y - self.dimensions[1] / 2. * self.scale.y
-        ymax = self.center.y + self.dimensions[1] / 2. * self.scale.y
-        return u.Quantity([ymin, ymax])
+        """
+        Return the Latitude (Y) range of the image from edge to edge.
+
+        *Note:* This does not imply that the values are linear along this
+         range, this just calculates the latitude values for the first and
+         last pixel.
+        """
+        cmin = self.pixel_to_data(0*u.pix, 0*u.pix)
+        lon_min, lat_min = self._get_lon_lat(cmin)
+        cmax = self.pixel_to_data(*self.dimensions)
+        lon_max, lat_max = self._get_lon_lat(cmax)
+        return u.Quantity([lat_min, lat_max])
```

Comment:
It's also not giving the same answers as the old implementation so watch this space.